ipython/Enwiki Revert Classifier.ipynb

```diff
@@ -0,0 +1,301 @@
+      "        parts = line.strip().split(\"\\t\")\n",
+      "        values = parts[:-1]\n",
+      "        reverted = parts[-1] == \"True\"\n",
+      "        \n",
+      "        feature_values = []\n",
+      "        for feature, value in zip(features, values):\n",
+      "            \n",
+      "            if feature.returns == bool:\n",
+      "                feature_values.append(value == \"True\")\n",
+      "            else:\n",
+      "                feature_values.append(feature.returns(value))\n",
+      "    \n",
+      "        feature_scores.append((feature_values, reverted))\n",
+      "\n",
+      "len(feature_scores)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": [
+      {
+       "metadata": {},
+       "output_type": "pyout",
+       "prompt_number": 2,
+       "text": [
+        "4982"
+       ]
+      }
+     ],
+     "prompt_number": 2
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## Training the models\n",
+      "\n",
+      "Now we'll use the extracted features to train some models.  For this section, I'll use an RBF and Linear kernel Support Vector Classifiers."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "from ores.models.enwiki import linear_svc, rbf_svc\n",
+      "from random import shuffle\n",
+      "\n",
+      "shuffle(feature_scores) # Randomize the set\n",
+      "\n",
+      "train_set = feature_scores[1000:] # Nearly 4000 training observations\n",
+      "test_set = feature_scores[:1000]  # 1000 test observations\n",
+      "\n",
+      "print(linear_svc.train(train_set))\n",
+      "print(rbf_svc.train(train_set))"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": [
+      {
+       "output_type": "stream",
+       "stream": "stdout",
+       "text": [
+        "{'seconds_elapsed': 13.189335584640503}\n",
+        "{'seconds_elapsed': 4.040912389755249}"
+       ]
+      },
+      {
+       "output_type": "stream",
+       "stream": "stdout",
+       "text": [
+        "\n"
+       ]
+      }
+     ],
+     "prompt_number": 3
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## Testing the models\n",
+      "\n",
+      "First, the linear kernel."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "import matplotlib.pyplot as plt\n",
+      "\n",
+      "stats = linear_svc.test(test_set)\n",
+      "print(\"mean.accuracy = {0}\".format(stats['mean.accuracy']))\n",
+      "print(\"auc = {0}\".format(stats['auc']))\n",
+      "\n",
+      "\n",
+      "plt.figure()\n",
+      "plt.plot(stats['roc']['fpr'],\n",
+      "         stats['roc']['tpr'], \n",
+      "         label='ROC curve')\n",
+      "plt.plot([0, 1], [0, 1], 'k--')\n",
+      "plt.xlabel('False Positive Rate')\n",
+      "plt.ylabel('True Positive Rate')\n",
+      "plt.show()"
```

Comment:
The figure should appear if you run it via ipython notebook or the viewer.  E.g. http://nbviewer.ipython.org/github/halfak/Objective-Revision-Evaluation-Service/blob/enwiki_accuracy/ipython/Enwiki%20Revert%20Classifier.ipynb